packages/webdriverio/src/middlewares.ts

```diff
@@ -24,9 +25,17 @@ export const elementErrorHandler = (fn: Function) => (commandName: string, comma
                  * as `stale element reference`
                  */
                 if (result && result.error === 'no such element') {
-                    const err = new Error()
-                    err.name = 'stale element reference'
-                    throw err
+                    let currentElement = element
+                    //Crawl back to the browser object
+                    while (currentElement.elementId && currentElement.parent) {
+                        currentElement = currentElement.parent as WebdriverIO.Element
+                    }
+                    const capabilities = currentElement?.capabilities as Capabilities.Capabilities
+                    if (capabilities?.browserName?.includes('safari')) {
```

Comment:
Why do we only do this for Safari browser? We should rather look into the protocol response and see how it differs from other browser to avoid have browser specific logic everywhere.

Reply:
Code that rewrites error name was added here https://github.com/webdriverio/webdriverio/pull/3701#discussion_r263853285 for Safari related bug https://github.com/SeleniumHQ/selenium/issues/8376. Before my changes it worked for all browsers instead of working only for Safari.
The bug is still open, and there is no difference between Safari's `not found element` and Safari's `stale element exception.`

Reply:
It seems like throwing the error here is causing the whole issue in the first place. It is independent from the browser. It might make sense to see why this was put here in the first place and whether we want to consider removing this piece afterall.

Reply:
Yes, current issue is independent from the browser. But it's causing by code that was added for handling  [the issue in  SafariDriver](https://bugs.webkit.org/show_bug.cgi?id=216479)
If we remove it we won't be able to handle `stale element exception` (that looks like `not found element`) in Safari anymore, .It can break some tests.